pyproject.toml

```
@@ -1,22 +1,79 @@
 [build-system]
 requires = ["setuptools>=75.6.0", "wheel", "setuptools_scm[toml]>=5.0"]
+build-backend = "setuptools.build_meta"
 
-[tool.mypy]
-exclude = "build/"
-plugins = ["pydantic.mypy"]
+[project]
+name = "evm-trace"
+description = "evm-trace: Ethereum Virtual Machine transaction tracing tool"
+readme = "README.md"
+authors = [{ name = "ApeWorX Ltd.", email = "admin@apeworx.io" }]
+license = { text = "Apache-2.0" }
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: Apache Software License",
+    "Natural Language :: English",
+    "Operating System :: MacOS",
+    "Operating System :: POSIX",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+    "Programming Language :: Python :: 3.14",
+]
+requires-python = ">=3.10,<4"
+dependencies = [
+    "cchecksum>=0.0.3,<1",
+    "eth-pydantic-types>=0.2.0,<0.3",
+    "eth-utils>=2.3.1,<6",
+    "msgspec>=0.8",
+    "pydantic>=2.5.2,<3",
+    "py-evm>=0.10.1b1,<0.13",
+]
+dynamic = ["version"]
+
+[project.urls]
+Homepage = "https://github.com/ApeWorX/evm-trace"
+
+[project.optional-dependencies]
```

Use dependency groups instead
```suggestion
[project.dependency-groups]
```